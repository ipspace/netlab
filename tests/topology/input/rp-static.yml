---
provider: clab
defaults.device: none

addressing.p2p.ipv6: 2001:db8:1::/48
addressing.loopback.ipv6: 2001:db8::/48

prefix:
  p1:
    ipv4: 10.42.0.0/16
  p2:
    ipv6: 2001:db8:cafe:1::/64

module: [ vrf ]
vrfs:
  red:
    links: [ c-p ]

routing.static:
  lan_pool:
    pool: lan
    nexthop.node: x

nodes:
  c:
    module: [ routing, vrf ]
    routing.static:
    - ipv4: 0.0.0.0/0         # Expand into NH-list
      nexthop.node: p
    - pool: loopback          # Checking pool prefix and remote nexthop
      nexthop.node: x
    - node: x                 # Checking static route to node CP endpoint
      nexthop.node: p
    - prefix: p1              # Static route to a named prefix
      nexthop.ipv4: 10.0.0.1
    - prefix: p2              # Named prefix, remote CP endpoint
      nexthop.node: x
    - include: lan_pool       # Included static route
      nexthop.ipv4: 10.0.0.42
    - ipv4: 172.16.0.0/16     # Intra-VRF static route
      vrf: red
      nexthop.node: p
    - ipv4: 172.17.0.0/16     # VRF-to-global static route
      vrf: red
      nexthop:
        vrf:
        node: p
    - ipv4: 172.18.0.0/16     # Global-to_VRF static route
      nexthop:
        vrf: red
        node: p
  p:
  x:

links:
- c-p
- c-p
- p-x
