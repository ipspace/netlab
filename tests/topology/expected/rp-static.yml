input:
- topology/input/rp-static.yml
- package:topology-defaults.yml
links:
- _linkname: links[1]
  interfaces:
  - ifindex: 1
    ifname: eth1
    ipv4: 10.1.0.1/30
    ipv6: 2001:db8:1::1/64
    node: c
  - ifindex: 1
    ifname: eth1
    ipv4: 10.1.0.2/30
    ipv6: 2001:db8:1::2/64
    node: p
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 10.1.0.0/30
    ipv6: 2001:db8:1::/64
  type: p2p
- _linkname: links[2]
  interfaces:
  - ifindex: 2
    ifname: eth2
    ipv4: 10.1.0.5/30
    ipv6: 2001:db8:1:1::1/64
    node: c
  - ifindex: 2
    ifname: eth2
    ipv4: 10.1.0.6/30
    ipv6: 2001:db8:1:1::2/64
    node: p
  linkindex: 2
  node_count: 2
  prefix:
    ipv4: 10.1.0.4/30
    ipv6: 2001:db8:1:1::/64
  type: p2p
- _linkname: links[3]
  interfaces:
  - ifindex: 3
    ifname: eth3
    ipv4: 10.1.0.9/30
    ipv6: 2001:db8:1:2::1/64
    node: p
  - ifindex: 1
    ifname: eth1
    ipv4: 10.1.0.10/30
    ipv6: 2001:db8:1:2::2/64
    node: x
  linkindex: 3
  node_count: 2
  prefix:
    ipv4: 10.1.0.8/30
    ipv6: 2001:db8:1:2::/64
  type: p2p
- _linkname: vrfs.red.links[1]
  interfaces:
  - ifindex: 3
    ifname: eth3
    ipv4: 10.1.0.13/30
    ipv6: 2001:db8:1:3::1/64
    node: c
  - ifindex: 4
    ifname: eth4
    ipv4: 10.1.0.14/30
    ipv6: 2001:db8:1:3::2/64
    node: p
  linkindex: 4
  node_count: 2
  prefix:
    ipv4: 10.1.0.12/30
    ipv6: 2001:db8:1:3::/64
  type: p2p
  vrf: red
module:
- routing
- vrf
name: input
nodes:
  c:
    af:
      ipv4: true
      ipv6: true
      vpnv4: true
      vpnv6: true
    box: none
    device: none
    hostname: clab-input-c
    id: 1
    interfaces:
    - ifindex: 1
      ifname: eth1
      ipv4: 10.1.0.1/30
      ipv6: 2001:db8:1::1/64
      linkindex: 1
      name: c -> p
      neighbors:
      - ifname: eth1
        ipv4: 10.1.0.2/30
        ipv6: 2001:db8:1::2/64
        node: p
      type: p2p
    - ifindex: 2
      ifname: eth2
      ipv4: 10.1.0.5/30
      ipv6: 2001:db8:1:1::1/64
      linkindex: 2
      name: c -> p
      neighbors:
      - ifname: eth2
        ipv4: 10.1.0.6/30
        ipv6: 2001:db8:1:1::2/64
        node: p
      type: p2p
    - ifindex: 3
      ifname: eth3
      ipv4: 10.1.0.13/30
      ipv6: 2001:db8:1:3::1/64
      linkindex: 4
      name: c -> p
      neighbors:
      - ifname: eth4
        ipv4: 10.1.0.14/30
        ipv6: 2001:db8:1:3::2/64
        node: p
        vrf: red
      type: p2p
      vrf: red
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.1/32
      ipv6: 2001:db8:0:1::1/64
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    module:
    - routing
    - vrf
    name: c
    routing:
      static:
      - ipv4: 10.0.0.0/24
        ipv6: 2001:db8::/48
        nexthop:
          idx: 0
          ipv4: 10.0.0.3
          ipv6: 2001:db8:0:3::1
          node: x
        pool: loopback
      - ipv4: 10.42.0.0/16
        nexthop:
          idx: 0
          ipv4: 10.0.0.1
        prefix: p1
      - ipv6: 2001:db8:cafe:1::/64
        nexthop:
          idx: 0
          ipv4: 10.0.0.3
          ipv6: 2001:db8:0:3::1
          node: x
        prefix: p2
      - ipv4: 172.16.0.0/16
        nexthop:
          idx: 0
          ipv4: 10.0.0.42
          ipv6: 2001:db8:0:3::1
          node: x
        pool: lan
      - ipv4: 0.0.0.0/0
        nexthop:
          idx: 0
          intf: eth1
          ipv4: 10.1.0.2
          ipv6: 2001:db8:1::2
      - ipv4: 0.0.0.0/0
        nexthop:
          idx: 1
          intf: eth2
          ipv4: 10.1.0.6
          ipv6: 2001:db8:1:1::2
      - ipv4: 10.0.0.3/32
        nexthop:
          idx: 0
          intf: eth1
          ipv4: 10.1.0.2
          ipv6: 2001:db8:1::2
      - ipv4: 10.0.0.3/32
        nexthop:
          idx: 1
          intf: eth2
          ipv4: 10.1.0.6
          ipv6: 2001:db8:1:1::2
      - ipv6: 2001:db8:0:3::/64
        nexthop:
          idx: 0
          intf: eth1
          ipv4: 10.1.0.2
          ipv6: 2001:db8:1::2
      - ipv6: 2001:db8:0:3::/64
        nexthop:
          idx: 1
          intf: eth2
          ipv4: 10.1.0.6
          ipv6: 2001:db8:1:1::2
      - ipv4: 172.16.0.0/16
        nexthop:
          idx: 0
          intf: eth3
          ipv4: 10.1.0.14
          ipv6: 2001:db8:1:3::2
        vrf: red
      - ipv4: 172.17.0.0/16
        nexthop:
          idx: 0
          intf: eth1
          ipv4: 10.1.0.2
          ipv6: 2001:db8:1::2
          vrf: null
        vrf: red
      - ipv4: 172.17.0.0/16
        nexthop:
          idx: 1
          intf: eth2
          ipv4: 10.1.0.6
          ipv6: 2001:db8:1:1::2
          vrf: null
        vrf: red
      - ipv4: 172.18.0.0/16
        nexthop:
          idx: 0
          intf: eth3
          ipv4: 10.1.0.14
          ipv6: 2001:db8:1:3::2
          vrf: red
    vrf:
      as: 65000
    vrfs:
      red:
        af:
          ipv4: true
          ipv6: true
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
  p:
    af:
      ipv4: true
      ipv6: true
      vpnv4: true
      vpnv6: true
    box: none
    device: none
    hostname: clab-input-p
    id: 2
    interfaces:
    - ifindex: 1
      ifname: eth1
      ipv4: 10.1.0.2/30
      ipv6: 2001:db8:1::2/64
      linkindex: 1
      name: p -> c
      neighbors:
      - ifname: eth1
        ipv4: 10.1.0.1/30
        ipv6: 2001:db8:1::1/64
        node: c
      type: p2p
    - ifindex: 2
      ifname: eth2
      ipv4: 10.1.0.6/30
      ipv6: 2001:db8:1:1::2/64
      linkindex: 2
      name: p -> c
      neighbors:
      - ifname: eth2
        ipv4: 10.1.0.5/30
        ipv6: 2001:db8:1:1::1/64
        node: c
      type: p2p
    - ifindex: 3
      ifname: eth3
      ipv4: 10.1.0.9/30
      ipv6: 2001:db8:1:2::1/64
      linkindex: 3
      name: p -> x
      neighbors:
      - ifname: eth1
        ipv4: 10.1.0.10/30
        ipv6: 2001:db8:1:2::2/64
        node: x
      type: p2p
    - ifindex: 4
      ifname: eth4
      ipv4: 10.1.0.14/30
      ipv6: 2001:db8:1:3::2/64
      linkindex: 4
      name: p -> c
      neighbors:
      - ifname: eth3
        ipv4: 10.1.0.13/30
        ipv6: 2001:db8:1:3::1/64
        node: c
        vrf: red
      type: p2p
      vrf: red
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.2/32
      ipv6: 2001:db8:0:2::1/64
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    module:
    - vrf
    name: p
    vrf:
      as: 65000
    vrfs:
      red:
        af:
          ipv4: true
          ipv6: true
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
  x:
    af:
      ipv4: true
      ipv6: true
    box: none
    device: none
    hostname: clab-input-x
    id: 3
    interfaces:
    - ifindex: 1
      ifname: eth1
      ipv4: 10.1.0.10/30
      ipv6: 2001:db8:1:2::2/64
      linkindex: 3
      name: x -> p
      neighbors:
      - ifname: eth3
        ipv4: 10.1.0.9/30
        ipv6: 2001:db8:1:2::1/64
        node: p
      type: p2p
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.3/32
      ipv6: 2001:db8:0:3::1/64
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.103
      mac: 08:4f:a9:00:00:03
    module: []
    name: x
    vrf:
      as: 65000
prefix:
  any:
    ipv4: 0.0.0.0/0
    ipv6: ::/0
  p1:
    _name: p1
    ipv4: 10.42.0.0/16
  p2:
    _name: p2
    ipv6: 2001:db8:cafe:1::/64
provider: clab
routing:
  static:
    lan_pool:
    - nexthop:
        node: x
      pool: lan
vrf:
  as: 65000
vrfs:
  red:
    export:
    - '65000:1'
    id: 1
    import:
    - '65000:1'
    rd: '65000:1'
